java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/service/impl/MenuServiceImpl.java

```diff
@@ -1,52 +1,90 @@
 package com.xxAMIDOxx.xxSTACKSxx.service.impl;
 
 import com.microsoft.azure.spring.data.cosmosdb.core.query.CosmosPageRequest;
+import com.xxAMIDOxx.xxSTACKSxx.api.v1.menu.dto.SearchMenuResultItem;
 import com.xxAMIDOxx.xxSTACKSxx.model.Menu;
 import com.xxAMIDOxx.xxSTACKSxx.repository.MenuRepository;
 import com.xxAMIDOxx.xxSTACKSxx.service.MenuService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Service;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 public class MenuServiceImpl implements MenuService {
 
     private static Logger logger = LoggerFactory.getLogger(MenuServiceImpl.class);
 
-    private final MenuRepository menuRepository;
+    private MenuRepository menuRepository;
 
     public MenuServiceImpl(MenuRepository menuRepository) {
         this.menuRepository = menuRepository;
     }
 
-    public List<Menu> all(int pageNumber, int pageSize) {
+    public List<SearchMenuResultItem> all(int pageNumber, int pageSize) {
```

Comment:
Loads of duplication in this class.
Also, it's the `MenuService`, not the `SearchMenuItemService` so should be returning `List<Menu>` not `List<SearchMenuResultItem>`